libcommon/chfn_checks.c

```diff
@@ -96,6 +96,34 @@ mbstowcs_alloc (const char *string, wchar_t **ret)
   if (!string)
     return -EINVAL;
 
+  /*
+   * it seems this spot here can break if `mbstowcs` is not happy with the
+   * encoding or something.
+   *
+   * I provided this input to org.openSUSE.pwupd.Chfn
+   *
+   * {
+   *         "userName": "mgerstner",
+   *         "homePhone": "stuff;böp"
+   * }
+   *
+   * this resulted in:
+   *
+   *    Nov 20 10:29:35 localhost.localdomain systemd-coredump[2453]:
```

Comment:
I can reproduce this crash with the chfn frontend itself (when adding the check there afterwards I forgot another call), but not with pwupdd, that still passes my tests with UTF-8.
The crash itself is now fixed (missing error check), but this will only fix the crash, not that the function fails.
@mgerstner Which locale are you using?

Reply:
[chauthtok.py](https://github.com/user-attachments/files/23673159/chauthtok.py)

Nothing special really. Default install of Tumbleweed:

```
$ locale
LANG=en_US.utf8
LC_CTYPE="en_US.utf8"
LC_NUMERIC="en_US.utf8"
LC_TIME=de_DE.utf8
LC_COLLATE="en_US.utf8"
LC_MONETARY="en_US.utf8"
LC_MESSAGES="en_US.utf8"
LC_PAPER="en_US.utf8"
LC_NAME="en_US.utf8"
LC_ADDRESS="en_US.utf8"
LC_TELEPHONE="en_US.utf8"
LC_MEASUREMENT="en_US.utf8"
LC_IDENTIFICATION="en_US.utf8"
LC_ALL=
```

I've attached the Python script that I'm using for this, maybe Python does something special here when calling `encode()` on the string.